src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserControllerIntTest.kt

```diff
@@ -15,6 +15,70 @@ import uk.gov.justice.digital.hmpps.manageusersapi.integration.IntegrationTestBa
 
 class UserControllerIntTest : IntegrationTestBase() {
 
+  @Nested
+  inner class AmendUserEmail {
+    private val username = "testy"
+    private val domain = "testing.com"
+    private val newEmailAddress = "new.testy@$domain"
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/prisonusers/$username/email")
+        .body(
+          fromValue(
+            mapOf("email" to newEmailAddress),
+          ),
+        )
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/prisonusers/$username/email")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(
+          fromValue(
+            mapOf("email" to newEmailAddress),
+          ),
+        )
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.post().uri("/prisonusers/$username/email")
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG_ROLE")))
+        .body(
+          fromValue(
+            mapOf("email" to newEmailAddress),
+          ),
+        )
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `amend user email`() {
+      nomisApiMockServer.stubFindUserByUsername(username)
+      hmppsAuthMockServer.stubConfirmRecognised(username)
+      hmppsAuthMockServer.stubUpdatePrisonUserEmail(username, newEmailAddress)
+      externalUsersApiMockServer.stubValidateEmailDomain(domain, true)
+      hmppsAuthMockServer.stubForTokenByEmailType()
+
+      webTestClient.post().uri("/prisonusers/$username/email")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .body(
+          fromValue(
+            mapOf("email" to newEmailAddress),
+          ),
+        )
+        .exchange()
+        .expectStatus().isOk
```

Comment:
Do we need tests to ensure we fail gracefully for failures - also - can an empty link ever be returned when not in a smoke test

Reply:
Discussed and have a added integration test case to ensure that remote failure passed on to client.